RasterHeader for storing raster meta data

rinkk opened the pull request:
this is the first follow-up on #1004 (and should be merged *after* #1004)
it introduces a tiny struct RasterHeader to story raster meta data and uses this throughout ogs where raster data is handled. it basically replaces 6 parameters in method-calls with one. it also fixes an index bug in Raster to VtkImageData conversion.

wenqing commented on Applications/DataExplorer/VtkVis/VtkRaster.cpp:
```diff
@@ -94,10 +86,10 @@ vtkImageImport* VtkRaster::loadImageFromArray(double const*const data_array, dou
+		image->SetDataSpacing(header.cell_size, header.cell_size, header.cell_size);
+		image->SetDataOrigin(header.origin[0]+(header.cell_size/2.0), header.origin[1]+(header.cell_size/2.0), 0);	// translate whole mesh by half a pixel in x and y
+		image->SetWholeExtent(0, header.n_rows-1, 0, header.n_cols-1, 0, 0);
+		image->SetDataExtent(0, header.n_rows-1, 0, header.n_cols-1-1, 0, 0);
```
header.n_cols-1-1 -->header.n_cols - 2

TomFischer commented on Applications/DataExplorer/VtkVis/VtkVisPipelineView.cpp:
```diff
@@ -196,13 +196,15 @@ void VtkVisPipelineView::addPipelineFilterItem()
 }
 
-void VtkVisPipelineView::constructMeshFromImage(QString msh_name, MeshLib::MeshElemType element_type, MeshLib::UseIntensityAs intensity_type)
+void VtkVisPipelineView::constructMeshFromImage(std::string msh_name, MeshLib::MeshElemType element_type, MeshLib::UseIntensityAs intensity_type)
```
Why not `std::string const & msh_name`?

TomFischer commented on GeoLib/Raster.h:
```diff
@@ -58,32 +61,35 @@ class Raster {
 	 * @return the origin of the raster
 	 */
-	GeoLib::Point const& getOrigin() const;
+	GeoLib::Point const getOrigin() const;
```
Why not return a `const &` here?

TomFischer commented:
Good idea! :+1:

endJunction commented:
The changes in the last commit 'rebase' a785ff8 revert some changes in the same pr. Is it intended?

Also check for trailing whitespaces.... It would be nice if the code base stays in consistent format.

endJunction commented on Applications/DataExplorer/DataView/NetCdfConfigureDialog.cpp:
```diff
@@ -352,13 +354,14 @@ void NetCdfConfigureDialog::createDataObject()
 	double origin_x = (originLon < lastLon) ? originLon : lastLon;
 	double origin_y = (originLat < lastLat) ? originLat : lastLat;
-	double originNetCdf[3] = {origin_x, origin_y, 0};
+	MathLib::Point3d origin(std::array<double,3>{{origin_x, origin_y, 0}});
```
@TomFischer Would be nice if the Point3d could be directly initialized from an initializer list.

endJunction commented on Applications/DataExplorer/VtkVis/VtkRaster.cpp:
```diff
@@ -54,15 +54,8 @@ vtkImageAlgorithm* VtkRaster::loadImage(const std::string &fileName,
-						raster->getNoDataValue());
-	}
+	if (raster)
+		return VtkRaster::loadImageFromArray(raster->begin(), raster->getHeader());
```
Would you please explain how the `raster` is deleted after this return? It looks like a memory leak to me.

rinkk replied:
not part of this PR, i just did the syntactic changes for using RasterHeader.
i suppose i could change raster to a unique_ptr to solve the problem.

endJunction commented on FileIO/AsciiRasterInterface.cpp:
```diff
@@ -43,24 +43,21 @@ GeoLib::Raster* AsciiRasterInterface::getRasterFromASCFile(std::string const& fn
-		double* values = new double[n_cols*n_rows];
+	GeoLib::RasterHeader header;
+	if (readASCHeader(in, header)) {
+		double* values = new double[header.n_cols*header.n_rows];
```
It is better to use unique_ptr here.

rinkk replied:
We can do this later, it has nothing to do with this PR

endJunction commented on MeshLib/MeshGenerators/RasterToMesh.h:
```diff
@@ -0,0 +1,125 @@
+ * \file
+ * \author Karsten Rink
+ * \date   2011-08-23
+ * \brief  Definition of the VtkMeshConverter class.
```
Please syncronize the documentation with the file content.

endJunction commented:
There are two new files, RasterToMesh.{cpp,h} which look similar to ConvertRasterToMesh.cpp. Is it just a copy, or are there other changes?

rinkk commented:
ConvertRasterToMesh was a partial copy of VtkMeshConverter. Then VtkMeshConverter was changed which made the copy unnecessary. With the above changes I deleted ConvertRasterToMesh and split VtkMeshConverter into a mesh conversion part and an raster conversion part.

rinkk commented:
concerning the rebase: there was a merge conflict and I added the code from the current master at that position.

endJunction commented:
Looks good. :fast_forward: 

Regarding the 'rebase' commit, it only seemed strange to me that there were changes in the PR which were then removed while resolving a conflict. That could be an error.

Please rebase on current master and merge when tests are passing.
